File: 04122024_need_for_speed/fibonacci_sequence.jl
# Julia code to find the nth Fibonacci number
using BenchmarkTools
using Distributions

function fibonacci(n)
    if n <=1
        return n
    else
        return fibonacci(n-1) + fibonacci(n-2)
    end
end

num = 40

@time println("Fibonacci value: ", fibonacci(num))

File: 04122024_need_for_speed/fibonacci_sequence.R
# recursive function to generate Nth number of fibonacci sequence
fibonacci <- function(n) {
    if (n <= 1) {
        return(n)
    } else {
        return(fibonacci (n-1) + fibonacci (n-2))
    }
}

# which number you want to generate
num = 40

start_time <- Sys.time()

fibonacci(num)

end_time <- Sys.time()
print(paste0("Time Taken = ", end_time - start_time, " seconds"))